epitator/annospan.py

```diff
@@ -1,27 +1,33 @@
 #!/usr/bin/env python
 # coding=utf8
 from __future__ import absolute_import
-from lazy import lazy
+
+
+EMPTY_LIST = []
 
 
 class AnnoSpan(object):
     """
     A span of text with an annotation applied to it.
     """
-    def __repr__(self):
-        return u'{0}-{1}:{2}'.format(self.start, self.end, self.label)
+    __slots__ = ["start", "end", "doc", "metadata", "label", "base_spans"]
 
     def __init__(self, start, end, doc, label=None, metadata=None):
         self.start = start
         self.end = end
         self.doc = doc
         self.metadata = metadata
+        # Base spans is only non-empty on span groups.
+        self.base_spans = EMPTY_LIST
 
         if label is None:
             self.label = self.text
         else:
             self.label = label
 
+    def __repr__(self):
+        return u'AnnoSpan({0}-{1}, {2})'.format(self.start, self.end, self.label)
+
```

Comment:
Documents usually don't have a name property.

Reply:
Using variable names would be very tricky because the variable a name refers to differs by the scope you're in. The local variables with the scope of the __repr__ function would be different from those in the scope the print function is called from.

I don't think having the doc is that important here. The goal I'm after is to have a somewhat structured/compact description of the object that can be used in doc tests and make print statements more readable when debugging. The document can usually be inferred from prior log statements and including info from it in every span string would add a lot of line noise when lists of AnnoTier spans are printed.